packages/router-plugin/src/core/config.ts

```diff
@@ -94,9 +107,9 @@ export const configSchema = generatorConfigSchema.extend({
         .optional(),
     })
     .optional(),
-})
+}) satisfies z.ZodType<ConfigOptions>
 
-export const getConfig = (inlineConfig: Partial<Config>, root: string) => {
+export const getConfig = (inlineConfig: ConfigInput, root: string) => {
```

Comment:
This could be `unknown` yeah?

```suggestion
export const getConfig = (inlineConfig: unknown, root: string) => {
```